doc/user-manual/language/cubical.lagda.rst

```diff
@@ -97,10 +158,780 @@ of the interval ``i0`` and ``i1`` are the bottom and top elements, respectively
+        (d : P x refl) {y : A} (p : x ≡ y)
+      → P y p
+  J P d p = transport (λ i → P (p i) (λ j → p (i ∧ j))) d
+
+One subtle difference between paths and the propositional equality
+type of Agda is that the computation rule for ``J`` does not hold
+definitionally. If ``J`` is defined using pattern-matching as in the
+Agda standard library then this holds, however as the path types are
+not inductively defined this does not hold for the above definition of
+``J``. In particular, transport in a constant family is only the
+identity function up to a path which implies that the computation rule
+for ``J`` only holds up to a path:
+
+::
+
+  transportRefl : ∀ {ℓ} {A : Set ℓ} (x : A) → transport refl x ≡ x
+  transportRefl {A = A} x i = transp (λ _ → A) i x
+
+  JRefl : ∀ {ℓ} {A : Set ℓ} {x : A} (P : ∀ y → x ≡ y → Set ℓ)
+           (d : P x refl) → J P d refl ≡ d
+  JRefl P d = transportRefl d
+
+Internally in Agda the ``transp`` operation computes by cases on the
+type, so for example for Σ-types it is computed elementwise. For path
+types it is however not yet possible to provide the computation rule
+as we need some way to remember the endpoints of the path after
+transporting it. Furthermore, this must work for arbitrary higher
+dimensional cubes (as we can iterate the path types). For this we
+introduce the "homogeneous composition operations" (``hcomp``) that
+generalize binary composition of paths to n-ary composition of higher
+dimensional cubes.
+
+
+Partial elements and systems
+----------------------------
+
+In order to describe the homogeneous composition operations we need to
+be able to write partially specified n-dimensional cubes (i.e. cubes
+where some faces are missing). Given an element of the interval ``r :
+I`` there is a predicate ``IsOne`` which represents the constraint ``r
+= i1``. This comes with a proof that ``i1`` is in fact equal to ``i1``
+called ``1=1 : IsOne i1``. We use Greek letters like ``φ`` or ``ψ``
+when such an ``r`` should be thought of as being in the image of
+``IsOne``.
+
+Using this we introduce a type of partial elements called ``Partial φ
+A``, this is a special version of ``IsOne φ → A`` with a more
+extensional judgmental equality (two elements of ``Partial φ A`` are
+considered equal if they represent the same subcube, so the faces of
+the cubes can for example be given in different order and the two
+elements will still be considered the same). The idea is that
+``Partial φ A`` is the type of cubes in ``A`` that are only defined
+when ``IsOne φ``.  There is also a dependent version of this called
+``PartialP φ A`` which allows ``A`` to be defined only when ``IsOne
+φ``.
+
+.. code-block:: agda
+
+  Partial : ∀ {ℓ} → I → Set ℓ → Setω
+
+  PartialP : ∀ {ℓ} → (φ : I) → Partial φ (Set ℓ) → Setω
+
+Partial elements are introduced using pattern matching:
+
+::
+
+  sys : ∀ i → Partial (i ∨ ~ i) Set₁
+  sys i (i = i0) = Set
+  sys i (i = i1) = Set → Set
+
```

Comment:
You could use `A` instead of `Set₁` and `a_0` and `a_1` instead of `Set` and `Set → Set`, rather than these seemingly random terms. Or even just question marks instead of `Set` and `Set → Set`.

Reply:
Or maybe even better a concrete type like `Bool`